src/cli_plugin/install/downloaders/http.js

```diff
@@ -2,11 +2,47 @@ import Wreck from 'wreck';
 import Progress from '../progress';
 import { fromNode as fn } from 'bluebird';
 import { createWriteStream } from 'fs';
+import HttpProxyAgent from 'http-proxy-agent';
+import URL from 'url';
 
-function sendRequest({ sourceUrl, timeout }) {
+function getProxyAgent(sourceUrl, logger) {
+  const httpProxy = process.env.http_proxy || process.env.HTTP_PROXY;
+  // we have a proxy detected, lets use it
+  if (httpProxy) {
+    logger.log(`Picked up proxy ${httpProxy} from http_proxy environment variable.`);
+    // get the hostname of the sourceUrl
+    const hostname = URL.parse(sourceUrl).hostname;
+    const noProxy = process.env.no_proxy || process.env.NO_PROXY || '';
+    const excludedHosts = noProxy.split(',');
+
+    // proxy if the hostname is not in noProxy
+    const shouldProxy = (excludedHosts.indexOf(hostname) === -1);
```

Comment:
Done the first. For the second: `no_proxy` can have a rather complex syntax and lots of tools don't support it all. @tylersmalley and me discussed if we should make an easy implementation at the beginning or try to support it as good as possible and decided to just check for plain hostnames now. We don't support ports at the moment (e.g. `no_proxy="artifcacts.elastic.co:443"` wouldn't neither match. I would also count this trimming of spaces as "advanced" handling at the moment, since we need to process the all entries again. So we should discuss whether or not we would also enhance the `no_proxy` support in general then? Discuss plz :)